Upgrade jerry-debugger client to use python3

repasics opened the pull request:
Also upgrade pylint to use python3.

JerryScript-DCO-1.0-Signed-off-by: Csaba Repasi repasics@inf.u-szeged.hu

galpeter commented on jerry-debugger/jerry_client.py:
```diff
@@ -104,7 +104,7 @@ def do_next(self, args):
                         self.quit = True
                         return
-                    elif res_type == result.TEXT:
+                    if res_type == result.TEXT:
```
Why do we need to change `elif` to  `if`? This applies to most of the cases.

repasics replied:
Pylint does not accepts elif here becasue of return in previous if statement:

************* Module jerry_client
jerry-debugger/jerry_client.py:104:20: R1705: Unnecessary "elif" after "return" (no-else-return)

galpeter commented on jerry-debugger/jerry_client.py:
```diff
@@ -279,7 +279,6 @@ def main():
 
         protocol = Socket(address)
     elif args.protocol == "serial":
-        from jerry_client_serial import Serial
```
Do we really need to move this out of here? If so, we always expect that a user will install the serial package even if they will only use the socket based communication.

repasics replied:
Moved import to top level because of pylint.

************* Module jerry_client
jerry-debugger/jerry_client.py:281:8: C0415: Import outside toplevel (jerry_client_serial) (import-outside-toplevel)

repasics replied:
I can move back the import there and disable pylint on that row.

repasics replied:
This was only error locally so I reverted this.

galpeter commented on jerry-debugger/jerry_client_tcp.py:
```diff
@@ -29,7 +29,7 @@ def connect(self):
         The format of address depends on the address family.
         """
-        print("Connecting to: %s:%s" % (self.address[0], self.address[1]))
+        print(("Connecting to: %s:%s" % (self.address[0], self.address[1])))
```
Why do we need the extra parentheses?

galpeter commented on tools/runners/test262-harness.py:
```diff
@@ -405,6 +405,7 @@ def write(self, string):
         os.write(self.file_desc, string)
 
     def read(self):
+        # pylint: disable=undefined-variable
```
Which variable is undefined here?

ossy-szeged replied:
pylint with python3 thinks that file is undefined variable. But test262-harness.py is still python2 script. (will be changed soon)

galpeter replied:
Isn't this code just reading the file contents and returning? If yes the solution is simple: use "open" as a context manager read the data and return the contents.

ossy-szeged replied:
Yes. but it is completely unrelated to this PR.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -208,20 +212,20 @@ def __init__(self, is_func, byte_code_cp, source, source_name, line, column, nam
-            breakpoint = JerryBreakpoint(_line, offset, self)
-            self.lines[_line] = breakpoint
-            self.offsets[offset] = breakpoint
+            br_point = JerryBreakpoint(_line, offset, self)
```
Why do we need to rename the "breakpoint" variable? This also applies to other places.

repasics replied:
Rename is needed because breakopoint is a built-in.

************* Module jerry_client_main
jerry-debugger/jerry_client_main.py:215:12: W0622: Redefining built-in 'breakpoint' (redefined-builtin)

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -208,20 +212,20 @@ def __init__(self, is_func, byte_code_cp, source, source_name, line, column, nam
                   % (self.byte_code_cp, self.source_name, self.name, self.line, self.column))
 
-        result += ','.join([str(breakpoint) for breakpoint in self.lines.values()])
+        result += ','.join([str(br_point) for br_point in list(self.lines.values())])
```
Why do we need the `list` call here? the `.values()` should already be an iterable object.

rerobika replied:
this question is still open.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -381,12 +385,12 @@ def breakpoint_list(self):
             result += "=== %sActive breakpoints %s ===\n" % (self.green_bg, self.nocolor)
-            for breakpoint in self.active_breakpoint_list.values():
-                result += " %d: %s\n" % (breakpoint.active_index, breakpoint)
+            for br_point in list(self.active_breakpoint_list.values()):
```
Same as above:
* Why the variable rename
* Why the `list` call?

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -381,12 +385,12 @@ def breakpoint_list(self):
             result += "=== %sPending breakpoints%s ===\n" % (self.yellow_bg, self.nocolor)
-            for breakpoint in self.pending_breakpoint_list.values():
-                result += " %d: %s (pending)\n" % (breakpoint.index, breakpoint)
+            for br_point in list(self.pending_breakpoint_list.values()):
```
Same as above.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -398,13 +402,13 @@ def delete(self, args):
             return "Error: Breakpoint index expected\n" \
                    "Delete the given breakpoint, use 'delete all|active|pending' " \
                    "to clear all the given breakpoints\n "
-        elif args in ['all', 'pending', 'active']:
```
Why change the "elif" to "if" ?

rerobika replied:
There is a return above.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -418,20 +422,19 @@ def delete(self, args):
+        return "Error: Breakpoint %d not found\n" % (breakpoint_index)
 
-    def next(self):
+    def __next__(self):
```
This should not be changed. The `__next__` would mean that this is an iterator/iterable, but this is not an iterator but a command sent via the debugger connection.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -888,7 +891,7 @@ def _parse_source(self, data):
                 return ""
 
-            elif buffer_type in [JERRY_DEBUGGER_SOURCE_CODE, JERRY_DEBUGGER_SOURCE_CODE_END]:
+            if buffer_type in [JERRY_DEBUGGER_SOURCE_CODE, JERRY_DEBUGGER_SOURCE_CODE_END]:
```
Why change `elif` to `if`?

rerobika replied:
ditto.

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -980,34 +983,34 @@ def _parse_source(self, data):
-        for function in new_function_list.values():
-            for line, breakpoint in function.lines.items():
-                self.line_list.insert(line, breakpoint)
+        for function in list(new_function_list.values()):
+            for line, br_point in list(function.lines.items()):
```
Why do we need the list calls?

galpeter commented on jerry-debugger/jerry_client_main.py:
```diff
@@ -980,34 +983,34 @@ def _parse_source(self, data):
-                    if (src.source_name == breakpoint.source_name or
-                            src.source_name.endswith("/" + breakpoint.source_name) or
-                            src.source_name.endswith("\\" + breakpoint.source_name)):
+                for src in list(new_function_list.values()):
+                    if (src.source_name == br_point.source_name or
+                            src.source_name.endswith("/" + br_point.source_name) or
+                            src.source_name.endswith("\\" + br_point.source_name)):
                         source_lines = len(src.source)
                         break
```
Why do we need the `list` call here?

galpeter reviewed (changes requested):
General question: do we want to remove full python2 support?

repasics commented:
Updated the PR. 2to3 app changed the parenthesis, next function and list call.
I reverted the parenthesis- and 'next' function change, but had to keep 2 of list calls to avoid runtime errors.

lygstate commented:
this can be closed, python3 is already supported?

robertsipka commented:
I've just closed this PR due to python3 is already supported.
